### PR TITLE
fix(k8s): allow null in spec.files for deploy config

### LINKED ISSUE #4878: 0.13: [Bug]: kubernetes deploy action: crash when encountering falsy file entries
### Current Behavior

Garden crashes when encoutering a falsy (i.e., `""` or `null`) entry in the `spec.files` array of a kubernetes deploy action.

### Expected behavior

Garden should ignore the file.
This is a regression - it used to work in <0.13

### Reproducible example

```
# example.garden.yml
kind: Deploy
type: kubernetes
name: example
spec:
  files:
    - null
```

```
$ garden deploy example

Error processing config for Deploy action example:
Error validating configure handler output from provider kubernetes for Deploy type kubernetes: key .config does not match any of the allowed types. Available keys: config, supportedModes)
```

### Workaround

Not known.

### Additional context

This is required when using templating to determine whether a file is supposed to be included or not.
For example: 
```
kind: Deploy
type: kubernetes
name: example
spec:
  files:
    - "${var.ingress-enabled ? 'ingress.yaml' : null}"
```

### Your environment

* **OS:** arch linux
* **How I'm running Kubernetes:** GKE

`garden version`
0.13.9

### PR DESCRIPTION
**Pull request by shumailxyz**
**What this PR does / why we need it**:
ignore null values from spec.files in configure handler. Useful when some manifest files are to be used conditionally based on variable/template string resolution.

**Which issue(s) this PR fixes**:

Fixes #4878

**Special notes for your reviewer**:

### CODE REVIEW

**Review by Walther — COMMENTED** [review]
Nice work!
Could we have a regression test for this please 👉 👈

File: core/src/plugins/kubernetes/kubernetes-type/deploy.ts [hunk 1]
```diff
@@ -44,7 +44,8 @@ export const kubernetesDeployDefinition = (): DeployActionDefinition<KubernetesD
     configure: async ({ ctx, config }) => {
       // The `spec` field hasn't been validated here yet,
       // so the value of `files` might be `undefined` instead of a default empty array
-      let files = config.spec.files || []
+      // also we discard any falsy file values in files array
+      let files = (config.spec.files || []).filter((f) => !!f)
```
Comment (vvagaytsev) [inline core/src/plugins/kubernetes/kubernetes-type/deploy.ts:48]: We can use our fancy `isTruthy` helper here :)
```
let files = (config.spec.files || []).filter(isTruthy)
```

Comment (shumailxyz) [reply core/src/plugins/kubernetes/kubernetes-type/deploy.ts:48]: nice. I wasn't aware that this existed.

Comment (shumailxyz) [reply core/src/plugins/kubernetes/kubernetes-type/deploy.ts]: Updated.

**Review by vvagaytsev — APPROVED** [review]
Great, thanks for fixing this! 💯

**Comment by hnicke** [comment]
I'm rather impatiently waiting for this fix to get released :smiley:

**Comment by vvagaytsev** [comment]
@hnicke we have released [0.13.10](https://github.com/garden-io/garden/releases/tag/0.13.10)! :)

**Comment by hnicke** [comment]
great. thank you!

**Pull request merged**